app/src/main/java/com/example/tudeeapp/presentation/screen/task/TaskViewModel.kt

```diff
@@ -0,0 +1,189 @@
+        }
+    }
+
+    private fun updateUiStateWithFilters() {
+        val filteredTasks = filterTasks(
+            tasks = allTasks,
+            selectedDate = currentSelectedDate,
+            selectedStatus = currentSelectedStatus
+        )
+
+        val status = calculateStatusUiList(allTasks, currentSelectedDate, currentSelectedStatus)
+
+        _uiState.value = TaskUiState(
+            data = TasksUi(
+                calender = currentSelectedDate.createCalendarUi(
+                    getCurrentMonthYear(currentSelectedDate),
+                    daysOfMonth = formatDailyDate(
+                        getAllDaysOfCurrentMonth(currentSelectedDate)
+                    )
+                ),
+                status = status,
+                selectedStatus = currentSelectedStatus,
+                tasks = filteredTasks,
+                showDatePicker = _uiState.value.data.showDatePicker,
+                todayIndex = getSelectedDayIndex(currentSelectedDate)
+            ),
+            isLoading = false,
+            errorMessage = null
+        )
+    }
+
+    private fun filterTasks(
+        tasks: List<TaskItemUiState>,
+        selectedDate: LocalDate,
+        selectedStatus: TaskStatusUi
+    ): List<TaskItemUiState> {
+        return tasks.filter { task ->
+            val matchesDate = task.createdDate == selectedDate
+            val matchesStatus = task.status == selectedStatus
+            matchesDate && matchesStatus
+        }
+    }
+
+    private fun calculateStatusUiList(
+        tasks: List<TaskItemUiState>,
+        selectedDate: LocalDate,
+        selectedStatus: TaskStatusUi
+    ): List<StatusUi> {
+        val tasksForDate = tasks.filter { it.createdDate == selectedDate }
+        val groupedByStatus = tasksForDate.groupingBy { it.status }.eachCount()
+
+        return TaskStatusUi.entries.map { status ->
+            StatusUi(
+                name = status.status,
+                count = groupedByStatus[status] ?: 0,
+                isSelected = status == selectedStatus
+            )
+        }
+    }
+
+    private fun formatDailyDate(daysOfMonth: List<LocalDate>): List<DayUi> {
+        return daysOfMonth.map{ date->
+            DayUi(
+                name = date.dayOfWeek.name.lowercase().replaceFirstChar { it.uppercase() }.take(3),
+                num = date.dayOfMonth,
+                date = date
+            )
+        }
+    }
+
+    private fun getCurrentDate(): LocalDate {
+        return Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault()).date
+    }
+
+    private fun getCurrentMonthYear(date: LocalDate): String {
+        return "${date.month.name.lowercase().replaceFirstChar { it.uppercase() }.take(3)}, ${date.year}"
+    }
+
+    private fun getSelectedDayIndex(date: LocalDate): Int {
+        return getAllDaysOfCurrentMonth(date).indexOfFirst { it == currentSelectedDate }
+    }
+
+    private fun getAllDaysOfCurrentMonth(date: LocalDate): List<LocalDate> {
```

Comment:
```suggestion
        private fun getAllDaysOfCurrentMonth(date: LocalDate): List<LocalDate> {
        return (1..date.month.length(date.year)).map { day ->
            LocalDate(date.year, date.month, day)
        }
    }
```

what about using `date.month.length(date.year) ` directly from kotlinx.datetime, which internally handles all leap year calculations

Reply:
but this code is not working it required to use java.time.LocalDate